packages/react-devtools-shared/src/__tests__/profilingCache-test.js

```diff
@@ -844,6 +844,254 @@ describe('ProfilingCache', () => {
+
+    const ResizableComponent = React.memo(function ResizableComponent(props) {
+      // 0
+      React.useRef(null);
+      React.useContext(EditorContext);
+      React.useContext(DropTargetContext);
+      // useShowPropertyPane
+      // useDispatch
+      React.useContext(ReactReduxContext);
+      // 1
+      React.useCallback(() => {}, []);
+      // useWidgetSelection
+      // useDispatch
+      React.useContext(ReactReduxContext);
+      // 2
+      React.useCallback(() => {}, []);
+      // 3
+      React.useCallback(() => {}, []);
+      // useWidgetDragResize
+      // useDispatch
+      React.useContext(ReactReduxContext);
+      // 4
+      React.useCallback(() => {}, []);
+      // 5
+      React.useCallback(() => {}, []);
+      // useSelector
+      React.useContext(ReactReduxContext);
+      // 6
+      React.useRef();
+      // 7
+      React.useCallback();
+      // 8,9,10,11
+      const selectedState1 = useSyncExternalStoreWithSelector(
+        subscribe,
+        getState,
+        getState,
+        state => state.a,
+      );
+      React.useDebugValue(selectedState1);
+      // useSelector
+      React.useContext(ReactReduxContext);
+      // 12
+      React.useRef();
+      // 13
+      React.useCallback();
+      // 14,15,16,17
+      const selectedState2 = useSyncExternalStoreWithSelector(
+        subscribe,
+        getState,
+        getState,
+        state => state.a,
+      );
+      React.useDebugValue(selectedState2);
+      // useSelector
+      React.useContext(ReactReduxContext);
+      // 18
+      React.useRef();
+      // 19
+      React.useCallback();
+      // 20,21,22,23
+      const selectedState3 = useSyncExternalStoreWithSelector(
+        subscribe,
+        getState,
+        getState,
+        state => state.a,
+      );
+      React.useDebugValue(selectedState3);
+      // useSelector
+      React.useContext(ReactReduxContext);
+      // 24
+      React.useRef();
+      // 25
+      React.useCallback();
+      // 26,27,28,29
+      const valueThatChanges = useSyncExternalStoreWithSelector(
+        subscribe,
+        getState,
+        getState,
+        state => state.b,
+      );
+      React.useDebugValue(valueThatChanges);
+
+      return valueThatChanges;
+    });
+
+    utils.act(() => render(<ResizableComponent />));
+
+    utils.act(() => store.profilerStore.startProfiling());
+
+    utils.act(() => {
+      reduxStore = {...reduxStore, b: reduxStore.b + 1};
+      syncExternalStoreCallback();
+    });
+
+    utils.act(() => store.profilerStore.stopProfiling());
+
+    const rootID = store.roots[0];
+
+    const changeDescriptions = store.profilerStore
+      .getDataForRoot(rootID)
+      .commitData.map(commitData => commitData.changeDescriptions);
+
+    expect(changeDescriptions).toMatchInlineSnapshot(`
+      [
+        Map {
+          2 => {
+            "context": false,
+            "didHooksChange": true,
+            "hooks": [
+              31,
```

Comment:
We don't have that many hooks. Highest index is 29. It should probably be 28 since that's the index of the `React.useSyncExternalStore` call.